Sources/TuistKit/Services/ShareService.swift

```diff
@@ -131,29 +138,30 @@ struct ShareService {
             )
         }
 
-        if !apps.isEmpty, apps.allSatisfy({ $0.hasSuffix(".app") }) {
-            let appPaths = try apps.map {
-                try AbsolutePath(
-                    validating: $0,
-                    relativeTo: fileHandler.currentPath
+        let appPaths = try await apps.concurrentMap {
+            try AbsolutePath(
```

Comment:
What if I pass the name of an app to be read from the built products directory and a relative path to an archive? Won't that lead to a failure because `try Absolute` path will fail to resolve it?

Reply:
It will throw here: https://github.com/tuist/tuist/blob/main/Sources/TuistKit/Services/ShareService.swift#L158. Let me add a test for that.